### PR TITLE
Protects DOACTION() calls with Career():CANDOACTIONS

### LINKED ISSUE #22: Implement protections against career limits
I have a craft with antennas which I launched with the scripts here.

The script crashed when attempting to extend the antennas, because I do not yet have my VAB upgraded to permit additional action groups.

https://ksp-kos.github.io/KOS_DOC/general/career_limits.html describes the limits, and offers a test to protect :DOACTION() calls with Career():CANDOACTIONS.

### PR DESCRIPTION
**Pull request by mathuin**
Fixes #22.

This patch wraps all DOACTION() lines with a check for whether
the action is permitted at this stage of career mode.

The boot logic is a guess as that case has not occurred to me yet.

The lib_parts logic does work for me with a VAB that has not yet
been upgraded.

### CODE REVIEW

File: boot/boot.ks [hunk 1]
```diff
@@ -53,15 +53,20 @@ ELSE {
...
-		FOR P IN SHIP:PARTS {
-			IF P:MODULES:CONTAINS("ModuleDeployableAntenna") {
-				LOCAL M IS P:GETMODULE("ModuleDeployableAntenna").
-				FOR A IN M:ALLACTIONNAMES() {
-					IF A:CONTAINS("Extend") { M:DOACTION(A,True). }
-				}.
-			}
-		}.
-		REBOOT.
+		IF Career():CANDOACTIONS {
```
Comment (xeger) [inline boot/boot.ks:56]: Could simplify the code by calling `partsExtendAntennas` here instead of duplicating the for loops and the can-do checking logic. @fellipec, any downside to running `lib_parts` from inside the boot script?

**Review by xeger — COMMENTED** [review]
Looks good overall; I had one comment for @fellipec to consider, since he heavily modified the boot scripts. If he doesn't get back to us in a few days, I'll do some experiments and then merge.

Thanks for the contribution!

File: boot/boot.ks (hunk 1 above)
Comment (fellipec) [reply boot/boot.ks:56]: Unfortunatelly yes. The boot script still have not a copy of lib_parts (or any other library) on the ship, and if can't have a home connection it also can't switch to archive to run it. 

I'll try to test those files in next days, but they look fine.

**Comment by xeger** [comment]
Hmm; it seems like a real corner case if we have no KSC connection, cannot
raise antennas *and* have no scripts copied to any local volume -- but if
it's worth handling, then it's worth handling right. :-)

After the @mathuin change, we will fall through to the "I'm sorry Dave, I
can't do that" case at the bottom of the script. We can probably flash a
uiError message onto the screen so the user is aware what happened ... with
no way to recover, I guess the best we can do after that, is to turn SAS on
and halt execution!

Seem reasonable?

**Comment by fellipec** [comment]
Yes really a corner case that I judged be only possible with a poor ship design before know that in career mode we can have that issue. 

My suggestion is to flash a HUD message to the user knows what the problem is and also be aware that another problems may happen. For example, I think maneuver nodes need some kind of upgrade to be used.

**Comment by mathuin** [comment]
Yes, that page lists four capabilities which are enabled at certain points of the career path -- tracking unnamed objects, patched conics, making nodes, and performing actions.  The actions one seems to get me the most as it affects *everything* that's normally done by clicking on stuff.

I'm definitely down with a HUD message -- users should see that and figure it out quickly enough.

**Comment by xeger** [comment]
One final question for @mathuin and @fellipec: planning of nodes is principal to RAMP's operation; should I check whether nodes are enabled as well, during bootstrap, and throw a fatal error with HUD message if not? It might keep people from wasting their time debugging weird crashes when RAMP tries to `ADDNODE`.

**Comment by fellipec** [comment]
I agree. Im out my town so cant help but i agree with the checks giving feedback though hud.

Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by mathuin** [comment]
I also agree with throwing fatal errors when trying to create nodes if they are not yet unlocked.

**Comment by fellipec** [comment]
I think tuesday I can get this working

Obter o Outlook para Android<https://aka.ms/ghei36>

________________________________

**Comment by xeger** [comment]
I got the candoaction stuff working; for `CANMAKENODES` I just extended `circ` so it can work without nodes. That gets people into stable orbit at least.

I also made some extensive changes to the universal boot script; @fellipec or @mathuin would one of you mind having a quick look before I merge? https://github.com/xeger/kos-ramp/pull/24
